test/e2e/settings/no_auth_locations.go

```diff
@@ -51,16 +52,24 @@ var _ = framework.IngressNginxDescribe("No Auth locations", func() {
 		Expect(s).NotTo(BeNil())
 		Expect(s.ObjectMeta).NotTo(BeNil())
 
-		updateConfigmap(setting, noAuthPath, f.KubeClientSet)
-
 		bi := buildBasicAuthIngressWithSecondPath(host, f.Namespace.Name, s.Name, noAuthPath)
 		ing, err := f.EnsureIngress(bi)
 		Expect(err).NotTo(HaveOccurred())
 		Expect(ing).NotTo(BeNil())
+
+		if defaultNginxConfigMapData == nil {
```

Comment:
Why not just revert `updateConfigmap(setting, noAuthPath, f.KubeClientSet)` with `updateConfigmap(setting, "", f.KubeClientSet)` in `AfterEach` instead of introducing a new logic? If we want to be more restrictive we can also always reset configmap at https://github.com/kubernetes/ingress-nginx/blob/master/test/e2e/framework/framework.go#L107